### DISCUSSION TITLE
How to use version of deployed service for the related REST client dependency?

### DISCUSSION THREAD

**Discussion by kiwi-oss** [renovatebot/renovate#23301] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.7.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We use a Keycloak instance that is hosted and maintained by another team. To access the service from our Java code, we use a client via the `org.keycloak:keycloak-core` dependency. For the client to send correct requests  to the REST API, their version have to match.

Now the current version of Keycloak is a bit older and Renovate creates a PR to update to the latest client version.

We want to find a way to use the version of the running instance for the client dependency. How could we go about that? We can get the version via a request to the REST API of the service and maybe by looking it up in a file in a Git repository.

### What you tried so far.

I have seen in the documentation that there are [custom managers](https://docs.renovatebot.com/modules/manager/regex/) for detecting the dependency version and [custom data sources](https://docs.renovatebot.com/modules/datasource/custom) for checking the latest version via HTTP request.

However, the example shows use of the custom data source only in combination with a regex manager. In our case, we want to use the existing (Maven) manager and only change its data source for that one dependency.

**Comment by rarkins** [comment, marked as answer]
You could disable the dependency in the maven manager and then match it using the regex manager so that the custom data source can be used.

**Reply by kiwi-oss** [reply]
Thanks for the quick reply! This feels a bit like a hack, though. I hoped I don't have to write a regex for something that's already built-in.

I'll wait a bit to see if someone has an idea how to reuse the existing manager and if not, I'll mark this as the answer.

**Reply by kiwi-oss** [reply]
No other replies in two weeks, so I'm marking yours as the answer. Thanks again!
